reference/opt/shaders-msl/asm/frag/depth-array-texture-lod.lod-as-grad.1d-as-2d.agx-cube-grad.msl23.asm.frag

```diff
@@ -0,0 +1,45 @@
+#pragma clang diagnostic ignored "-Wmissing-prototypes"
+
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+static inline gradientcube spvGradientCube(float3 P, float3 dPdx, float3 dPdy)
+{
+    // Major axis selection
+    float3 absP = abs(P);
```

Comment:
I'll trust the math is correct.

Reply:
Actually, one question. What happens when the gradient for neighbor pixel cannot be projected onto the same face? E.g. lane 0 points to face +X, but lane 1 points to face -X. That should probably be considered "infinite" gradient?